game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/ClearBombardmentCasualties.java

```diff
@@ -0,0 +1,49 @@
+package games.strategy.triplea.delegate.battle.steps.change;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_BOMBARDMENT_CASUALTIES;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class ClearBombardmentCasualties implements BattleStep {
+  private static final long serialVersionUID = -5723287846470464298L;
+
+  private final BattleState battleState;
+
+  private final BattleActions battleActions;
+
+  @Override
+  public List<String> getNames() {
+    return valid() && clearCasualties() ? List.of(REMOVE_BOMBARDMENT_CASUALTIES) : List.of();
+  }
+
+  @Override
+  public BattleStep.Order getOrder() {
+    return Order.NAVAL_BOMBARDMENT_REMOVE_CASUALTIES;
+  }
+
+  @Override
+  public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+    if (valid() && clearCasualties()) {
+      battleActions.clearWaitingToDieAndDamagedChangesInto(bridge);
+    }
+  }
+
+  private boolean clearCasualties() {
+    return !Properties.getNavalBombardCasualtiesReturnFire(
+        battleState.getGameData().getProperties());
+  }
+
+  private boolean valid() {
```

Comment:
nit, valid could be more descriptive. When reading the usage, I wondered "what is valid?"
Some rename suggestions:
-  "isBombardmentRound"
- "bombardmentOccurred"
- "bombardUnitsWillFire"

Side-note, I wonder if it would make sense to create a `BombardmentRules` object so we can centralize logic like this, where we have one source  to detect, when/how/if bombardment should occur.

Reply:
I just wrote my thoughts on how to centralize the logic using the game XML in the forum.  You can see my thoughts at https://forums.triplea-game.org/topic/2481/custom-battle-phases.  That forum post encompasses more than just centralizing bombardment logic.